source/MaterialXCore/Unit.cpp

```diff
@@ -243,4 +243,70 @@ void UnitConverterRegistry::write(DocumentPtr doc) const
     }
 }
 
+bool UnitConverterRegistry::convertToUnit(DocumentPtr doc, const string& unitType, const string& targetUnit)
```

Comment:
This was a hack I put in for Arnold rendering to pre-convert anything with a unit in C++. I don't think this should go back as it only works in some cases.